providers/oidc.go

```diff
@@ -282,16 +282,23 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
 	groups := []string{}
 
-	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
-	if rawGroups != nil && ok {
+	switch rawGroups := rawClaims[p.GroupsClaim].(type) {
+	case []string:
+		groups = append(groups, rawGroups...)
+	case *[]string:
+		groups = append(groups, *rawGroups...)
+	case string:
+		groups = append(groups, rawGroups)
+	case *string:
+		groups = append(groups, *rawGroups)
+	case []interface{}:
```

Comment:
I was expecting us to only need the `[]string` and `string` cases. We should look at how this map is being generated but I had assumed it would be from the json unmarshaller, in which case, we should try to replicate this in the tests and verify which code paths are actually being walked here

Reply:
From what I found `rawGroup` was always `[]interface{}` that is why I ended up with the looping and casting, does this new version every hit any of the other cases?

Reply:
Thanks for chipping in, this is kind of what I was wondering as well. I wonder if there's a better way we can do this by unmarshalling into some clever type that can detect if it's a string or list of strings 🤔

Reply:
The pointer cases seem highly unlikely yes, I just added them as a reflex, removing them for now